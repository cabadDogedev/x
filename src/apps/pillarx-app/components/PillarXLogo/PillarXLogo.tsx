type PillarXLogoProps = {
  src: string;
  className?: string;
}

export const PillarXLogo = ({ src, className }: PillarXLogoProps) => {
  return (
    <img src={src} className={`w-min ${className}`} />
  )
}

export default PillarXLogo;
