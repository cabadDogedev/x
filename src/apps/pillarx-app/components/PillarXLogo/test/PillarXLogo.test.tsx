import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

// components
import PillarXLogo from '../PillarXLogo';

describe('<PillarXLogo />', () => {
    const src = 'https://example.com/logo.png';
  
    it('renders correctly and matches snapshot', () => {
      const tree = renderer
        .create(
          <>
            <PillarXLogo src={src} />
            <PillarXLogo src={src} className="custom-class" />
          </>
        )
        .toJSON();
  
      expect(tree).toMatchSnapshot();
    });
  
    it('renders the image with the correct src', () => {
      const tree = renderer.create(<PillarXLogo src={src} />).toJSON() as ReactTestRendererJSON;
      expect(tree.type).toBe('img');
      expect(tree.props.src).toBe(src);
    });
  
    it('applies the default class', () => {
      const tree = renderer.create(<PillarXLogo src={src} />).toJSON() as ReactTestRendererJSON;
      expect(tree.props.className).toContain('w-min');
    });
  
    it('applies the custom class', () => {
      const customClass = 'custom-class';
      const tree = renderer.create(<PillarXLogo src={src} className={customClass} />).toJSON() as ReactTestRendererJSON;
      expect(tree.props.className).toContain('w-min');
      expect(tree.props.className).toContain(customClass);
    });
  });
  