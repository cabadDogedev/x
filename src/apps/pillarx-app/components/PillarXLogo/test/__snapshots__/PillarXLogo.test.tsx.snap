// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PillarXLogo /> renders correctly and matches snapshot 1`] = `
[
  <img
    className="w-min undefined"
    src="https://example.com/logo.png"
  />,
  <img
    className="w-min custom-class"
    src="https://example.com/logo.png"
  />,
]
`;
